### DISCUSSION TITLE
PostgreSQL read replicas support

### DISCUSSION THREAD

**Discussion by linus-ha** [payloadcms/payload#7263] [Feature Requests & Ideas]
A typical configuration of PostgreSQL is running multiple DB Servers, where one is the main one and others are replicating the main one. The secondary Postgres instances are therefore in read-only. (Not using technical correct terms here)

Allowing access to these replicas allows for better read performance, splitting the work when querying (not mutating). Drizzle seems to have an implementation for this: https://orm.drizzle.team/docs/read-replicas

This should be fairly easy to implement. You should just need to allow the user setting replica pools in the payload config. Then, when replica pools are defined, init drizzle with `withReplicas`, as shown in the doc link I've provided.

**Comment by ZaheerHB** [comment]
Hello @linus-ha, have you tried implementing a custom solution for this?

I am also exploring a solution to implement a replica set like Master-Slave.

**Reply by linus-ha** [reply]
No, I have not yet. Just revisiting the topic right now. It seems like something like this got implemented in the meanwhile by Payload directly. Have not tested this yet.

**Comment by anyuj** [comment]
Hi @r1tsuu , any chance to support this in the future?

**Reply by frederichoule** [reply]
It's actually supported now: https://payloadcms.com/docs/database/postgres - although it's unusable because of https://github.com/payloadcms/payload/issues/13011

**Reply by linus-ha** [reply]
Thanks for the hint. Link to your last comment on that topic here so it's easy to find:

https://github.com/payloadcms/payload/pull/13023#issuecomment-3186905316
